charts/redpanda/templates/post-install-upgrade-job.yaml

```diff
@@ -76,23 +76,27 @@ spec:
 {{- if .Values.auth.sasl.enabled }}
   {{- $values := .Values }}
   {{- range $user := .Values.auth.sasl.users }}
-            # To avoid `set -e` from exiting the command when a user exists; catch the stderr output and exit codes into `creation_result`
-            # and `creation_result_exit_code` for use later
-            creation_result=$(rpk acl user create {{ $user.name }} -p {{ $user.password | quote }} --mechanism {{ include "sasl-user-mechanism" (dict "user" $user "Values" $values) }} {{ template "rpk-common-flags" $ }} 2>&1) && creation_result_exit_code=$? || creation_result_exit_code=$?
-
-            # On a non-success exit code
+    {{- if $user.password }}
+            PASSWORD={{ $user.password | quote }}
+    {{- else if not (empty $user.secretName) }}
+            PASSWORD=$(cat {{ printf "/etc/secret-name/%s/password" $user.secretName }})
+    {{- else }}
+            PASSWORD=""
+    {{- end }}
+            echo {{ printf "Creating user %q..." $user.name | quote }}
+            creation_result=$(rpk acl user create {{ $user.name }} --password=${PASSWORD} --mechanism {{ include "sasl-user-mechanism" (dict "user" $user "Values" $values) }} {{ template "rpk-common-flags-no-sasl" $ }} 2>&1) && creation_result_exit_code=$? || creation_result_exit_code=$?            # On a non-success exit code
             if [[ $creation_result_exit_code -ne 0 ]]; then
               # Check if the stderr contains "User already exists"
               if [[ $creation_result == *"User already exists"* ]]; then
-                printf "The %s user already exists, skipping creation.\n" {{ $user.name }}
+                echo {{ printf "The %s user already exists, skipping creation." $user.name | quote }}
```

Comment:
NIT: It would be good to add `rpk acl user update` command, but we can use curl to PATCH the password.

Reply:
Thanks, comment resolved 👍

Reply:
Quick question, do you know if `rpk acl user update` is idempotent? Can it also "create" a user even it it doesn't exist?